### PR TITLE
optional rcode in the report

### LINKED ISSUE #131: Not display SRC in the final report by default
Is it possible to not display the R code in the generated final report by default (in the pdf, html, word and pptx files)? The reason is the reports will be delivered to stakeholders, who are not programmers and would only need the analysis results instead of the source code. The app developer will still be able to retrieve the code from the Rmd to trace back or turn on the SRC in the Rmd if needed, maybe something like

```
```{r, results='hide'}
print("<R code here>")
```

### PR DESCRIPTION
**Pull request by Polkas**
closes #131 

blocked by https://github.com/insightsengineering/teal.reporter/pull/161

optional r code in the reporter document.
Using the global knitr options to transfer information if the r code should be echo-ed.

### CODE REVIEW

File: R/Previewer.R [hunk 1]
```diff
@@ -146,6 +141,15 @@ reporter_previewer_srv <- function(id, reporter, rmd_yaml_args = list(
           shiny::showNotification("Rendering and Downloading the document.")
           input_list <- lapply(names(rmd_yaml_args), function(x) input[[x]])
           names(input_list) <- names(rmd_yaml_args)
+          if (is.logical(input$showrcode)) {
```
Comment (Polkas) [inline R/Previewer.R:144]: Place to think about

File: R/DownloadModule.R [hunk 2]
```diff
@@ -134,6 +131,15 @@ download_report_button_srv <- function(id,
           shiny::showNotification("Rendering and Downloading the document.")
           input_list <- lapply(names(rmd_yaml_args), function(x) input[[x]])
           names(input_list) <- names(rmd_yaml_args)
+          if (is.logical(input$showrcode)) {
```
Comment (Polkas) [inline R/DownloadModule.R:134]: Place to think about

File: R/Renderer.R [hunk 3]
```diff
@@ -56,16 +59,24 @@ Renderer <- R6::R6Class( # nolint: object_name_linter.
...
+      )
+      parsed_blocks <- paste(
+        unlist(
+          lapply(blocks, function(b) private$block2md(b))
+        ),
+        collapse = "\n\n"
+      )
 
-      rmd_text <- paste0(parsed_yaml, "\n", parsed_blocks, "\n")
+      rmd_text <- paste0(parsed_yaml, "\n\n", parsed_global_knitr, "\n\n", parsed_blocks, "\n")
```
Comment (Polkas) [inline R/Renderer.R:79]:
```suggestion
      rmd_text <- paste0(parsed_yaml, "\n", parsed_global_knitr, "\n", parsed_blocks, "\n")
```

File: NEWS.md [hunk 4]
```diff
@@ -1,5 +1,11 @@
 # teal.reporter 0.1.1.9000
 
+### New features
+* Added additional input when needed to decided if Show R Code should be part of the report.
+
```
Comment (Polkas) [inline NEWS.md:5]:
```suggestion
### New features
* Added additional input when needed to decided if Show R Code should be part of the report.
* Added support for global knitr options in the render method in the Render class.

```

**Review by nikolas-burkoff — COMMENTED** [review]
A few comments from me

File: R/DownloadModule.R [hunk 5]
```diff
@@ -225,3 +226,35 @@ report_render_and_compress <- function(reporter, input_list, file = tempdir()) {
...
+}
+
+#' @keywords internal
+any_rcode_block <- function(reporter) {
+  length(
+    Filter(
+      function(e) inherits(e, "RcodeBlock"),
+      reporter$get_blocks()
+    )
+  ) > 0
```
Comment (Polkas) [inline R/DownloadModule.R:259]:
```suggestion
  any(
    Filter(
      function(e) inherits(e, "RcodeBlock"),
      reporter$get_blocks()
    )
  )
```

File: R/DownloadModule.R [hunk 6]
```diff
@@ -225,3 +226,35 @@ report_render_and_compress <- function(reporter, input_list, file = tempdir()) {
...
+          )
+        )
+      )
+    })
+  )
+}
+
+#' @keywords internal
+any_rcode_block <- function(reporter) {
+  length(
```
[the remark this answers is not in the document]
Comment (Polkas) [reply R/DownloadModule.R:254]: I made it as I work with list and there will be a conversion warning message. But I could use vapply.

File: R/DownloadModule.R [hunk 7]
```diff
@@ -97,17 +97,14 @@ download_report_button_srv <- function(id,
...
-            shinyWidgets::pickerInput(
-              inputId = ns("output"),
-              label = "Choose a document type: ",
-              choices = rmd_output,
-              selected = rmd_yaml_args$output
+          reporter_download_inputs(rmd_yaml_args),
+          if (any_rcode_block(reporter)) {
+            shiny::checkboxInput(
+              ns("showrcode"),
+              label = "R Code: echo=TRUE",
```
Comment (Polkas) [inline R/DownloadModule.R:104]:
```suggestion
              label = "Include R Code",
```

[the remark this answers is not in the document]
Comment (Polkas) [reply R/DownloadModule.R]: done

File: NEWS.md [hunk 8]
```diff
@@ -1,5 +1,12 @@
 # teal.reporter 0.1.1.9000
 
+### New features
+* Added additional input when needed to decided if Show R Code should be part of the report.
+* Added support for global knitr options in the render method in the Render class.
+
+### Breaking changes
+* Simplified `reporter_previewer_ui` to contain just `id` argument.
```
[the remark this answers is not in the document]
Comment (Polkas) [reply NEWS.md:8]: I check it now, Looks like not action needed there.

**Comment by Polkas** [comment]
@nikolas-burkoff linked with https://github.com/insightsengineering/teal.reporter/issues/165

File: R/Renderer.R [hunk 9]
```diff
@@ -57,16 +60,24 @@ Renderer <- R6::R6Class( # nolint: object_name_linter.
...
+    renderRmd = function(blocks, yaml_header, global_knitr = list()) {
       checkmate::assert_list(blocks, c("TextBlock", "PictureBlock", "NewpageBlock", "TableBlock", "RcodeBlock"))
       if (missing(yaml_header)) {
         yaml_header <- md_header(yaml::as.yaml(list(title = "Report")))
       }
-
       parsed_yaml <- yaml_header
-      parsed_blocks <- paste(unlist(lapply(blocks, function(b) private$block2md(b))), collapse = "\n\n")
+      parsed_global_knitr <- sprintf(
+        "\n```{r setup, include=FALSE}\noptions(width = 80)\nknitr::opts_chunk$set(%s)\n```\n",
```
Comment (Polkas) [inline R/Renderer.R:70]:
```suggestion
        "\n```{r setup, include=FALSE}\nknitr::opts_chunk$set(%s)\n```\n",
```

**Comment by Polkas** [comment]
@nikolas-burkoff I am confident now that the styling of the code should be done when it is appended to the report card.
We are printing content in the same shape when it was provided to us. This was a decision made in the beginning. 

Even if we want to continue the discussion then it is a new issue for me.

A small idea.
SO If sb add the 1000 chars line (not separated with new lines) and render a pdf document , obviously it will look bad.
Try to imagine sb want to have such long line in the html then we limit such usage.

**Comment by Polkas** [comment]
@nikolas-burkoff are you content about this PR:)?

**Review by nikolas-burkoff — APPROVED** [review]

**Pull request merged**